extension/src/Registry.ts

```diff
@@ -207,18 +222,26 @@ export class Registry {
     /**
      * Gets the version-specific metadata for a specific version of a package.
      *
-     * If `version` is omitted or `"latest"`, this returns the latest version.
+     * If `version` is the name of a release channel, this gets the latest version in that channel.
+     * If `version` is omitted, this gets the latest version for the user's selected channel.
+     * @throws VersionMissingError if the given version does not exist.
      */
-    public async getPackageVersionMetadata(name: string, version: string = 'latest') {
+    public async getPackageVersionMetadata(name: string, version?: string) {
         const metadata = await this.getPackageMetadata(name);
 
         assertType(metadata, PackageVersionData, `In package "${name}"`);
 
-        if (version in metadata['dist-tags']) {
-            version = metadata['dist-tags'][version];
+        if (version === undefined) {
+            const latest = lookupVersion(metadata, name, 'latest');
```

Comment:
Might be useful to leave a comment explaining that we need the publisher to look up the channel to use, but the publisher is only stored on version-specific metadata.